File: DSA/Gautam_Shyamnani_590013078/Day3Question3.cpp
#include<iostream>
#include<algorithm>
#include<vector>
using namespace std;
int secondlargest(vector<int>arr,int n){
    sort(arr.begin(),arr.end());
    int largest=arr[n-1];
    for(int i=n-2;i>=0;i--){
        if(arr[i]!=largest){
            int secondlargest=arr[i];
            return secondlargest;
        }
    }
    return -1;
}
int main(){
    vector<int>arr={9,8,8,18};
    int n=arr.size();
    cout<<secondlargest(arr,n)<<endl;
    return 0;
}

File: DSA/DivyanshKumarSingh_590012175/Day_3_Question_3.py
def secondLargest(lst):
    uNum = list(set(lst))  
    
    if len(uNum) < 2:
        return -1  
    
    uNum.sort(reverse=True)  
    return uNum[1]  


lst = list(map(int, input("Enter array elements separated by space: ").split()))
print("Second largest unique:", secondLargest(lst))
